lntest/itest/lnd_test.go

```diff
@@ -2895,6 +2921,465 @@ func testChannelBalance(net *lntest.NetworkHarness, t *harnessTest) {
 	closeChannelAndAssert(ctxt, t, net, net.Alice, chanPoint, false)
 }
 
+// testChannelFundMaxLessThanDust tries to create a new channel using fundmax
+// flag with a wallet balance less than dust. Should fail.
+func testChannelFundMaxLessThanDust(net *lntest.NetworkHarness, t *harnessTest) {
```

Comment:
Yay, nice to have these integration tests!

However, there are a few things to optimize here:
- It would be nice if we had one test for this that runs several sub test cases in a table driven manner. An example for this is `testChannelBackupRestore`.
- The two nodes `carol` and `dave` could be created once and reused for each (sub) test. To make sure they start with a zero balance before each test, you can use the `SendCoins` RPC with the `SendAll` flag to send all remaining balance to some address of the miner for example (see `testSweepAllCoins`). That should speed up things considerably as the nodes can keep running between the tests (the previously opened channels can remain open I think).
- We started moving new tests to their own files to make keep `lnd_test.go` from growing even further. Can you move the test to `lnd_channel-funding_fund_max.go` or something like that?
- Move everything related to the itest into a separate commit.